tests/phpunit/tests/rest-api/rest-posts-controller.php

```diff
@@ -749,6 +750,48 @@ public function test_get_items_status_without_permissions() {
 		}
 	}
 
+	/**
+	 * @ticket 56350
+	 */
+	public function test_get_items_exact_search() {
+
+		self::factory()->post->create(
+			array(
+				'post_title'   => 'Rye',
+				'post_content' => 'This is a post about Rye Bread',
+				'post_status'  => 'publish',
+			)
+		);
+
+		self::factory()->post->create(
+			array(
+				'post_title'   => 'Types of Bread',
+				'post_content' => 'Types of bread are White and Rye Bread',
+				'post_status'  => 'publish',
+			)
+		);
+
+		$request = new WP_REST_Request( 'GET', '/wp/v2/posts' );
+
+		// General search.
+		$request->set_param( 'search', 'Rye' );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		$this->assertCount( 2, $data, 'Querying the API without exact_search should return all posts containing the search keyword' );
+
+		// Exact search using same search param.
+		$request->set_param( 'exact_search', true );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		$this->assertCount( 1, $data, 'Querying the API with exact_search should return posts matching the search keyword' );
+
+		// Note that "exact_search" is still true.
+		$request->set_param( 'search', 'Rye Bread' );
```

Comment:
What result do you get when you search `Types of Bread`?

Reply:
Much like `/?s=Types+of+Bread&exact=1`, this will not return a match. This applies to any exact search in which the search term includes a space.

Reply:
When we exact search the one word, it will return the posts matching exactly, but if there are multiple words to exact search, it doesn't work actually. so would it make sense to introduce `sentence` as well in the REST api?

Reply:
@TimothyBJacobs What do you think?

Reply:
https://github.com/WordPress/wordpress-develop/pull/3677#issuecomment-2356523514